File: frontend/src/types/project.ts
export type ProjectStatus = "draft" | "active" | "archived";

export interface Project {
  id: string;
  name: string;
  description?: string;
  status: ProjectStatus;
  created_at: string;
  updated_at?: string;
}

export interface ProjectCreate {
  name: string;
  description?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  status?: ProjectStatus;
}

File: backend/src/models/repository/project.py
from datetime import datetime
from pydantic import BaseModel, Field
from ..base import BaseEntity, BaseSemanticModel


class Project(BaseEntity, BaseSemanticModel):
    status: str = Field(default="draft", description="Project status: draft, active, archived")


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: str | None = None
